backend/ipp.c

```diff
@@ -3624,4 +3624,7 @@ update_reasons(ipp_attribute_t *attr,	/* I - printer-state-reasons or NULL */
     fprintf(stderr, "%s\n", add);
   else if (rem[0])
     fprintf(stderr, "%s\n", rem);
+
+  if (new_reasons != NULL)
+    cupsArrayDelete(new_reasons);
```

Comment:
OK, we should just initialize new_reasons to NULL up top and then call cupsArrayDelete unconditionally (it handles NULLs just fine.)